datacube/index/eo3.py

```diff
@@ -273,3 +283,346 @@ def lineage_remap(name: str, uuids) -> dict[str, Any]:
+# landsat:processing_level  "L1GT"
+# landsat:tier  "RT"
+
+_EO1_PROPERTY_MAP = {
+    "platform": field_platform,
+    "instrument": field_instrument,
+    "sat_path": field_path_row,
+    "sat_row": field_path_row,
+    "label": field_label,
+    "format": field_format,
+    "product_type": field_product,
+}
+
+
+def _build_properties(d: DocReader):
+    for key, val in d.fields.items():
+        if val is None:
+            continue
+        converter = _EO1_PROPERTY_MAP.get(key)
+        if converter:
+            yield from converter(key, val)
+
+
+def make_grids(
+    ds: Dataset, open_datafiles: bool
+) -> tuple[dict[str, dict[str, Any]], dict[str, str]]:
+    """
+    Determine grid values if possible from geo_ref_points and cell_size/shape values.
+    If measurements do not contain the necessary information, see if it can be found in
+    browse or else the product definition.
+    If open_datafile is True, open the measurement datafiles and read the information from there.
+    This will be slow and should only be used as a last resort.
+    Return grids dict and (optionally) mapping of measurement name to grid name.
+    """
+    geoboxes = {}
+    assert ds._gs is not None
+    ref_points: dict = ds._gs["geo_ref_points"]
+
+    def _shape_and_transform(
+        res: int | float | dict | None,
+        shape: dict | None,
+        adjust: bool = False,
+    ) -> tuple[tuple[int, int], Affine] | tuple[None, None]:
+        """Calculate shape (y,x) and transform, or return None if they cannot be determined"""
+        if res:
+            if isinstance(res, int | float):
+                res_x, res_y = res, -res
+            else:
+                # y value is not always negative even when we are dealing with a square resolution
+                res_x, res_y = res["x"], -res["y"] if res["y"] == res["x"] else res["y"]
+            transform = Affine(
+                res_x, 0.0, ref_points["ul"]["x"], 0.0, res_y, ref_points["ul"]["y"]
+            )
+            shape_x, shape_y = ~transform * (
+                ref_points["lr"]["x"],
+                ref_points["lr"]["y"],
+            )
+        elif shape:
+            # the browse.full.shape values generally seem to be accurate but off by 1, so adjust accordingly
+            # this logic has been extrapolated from a very small sample size and may be removed in the future
+            shape_x, shape_y = (
+                (shape["x"] - 1, shape["y"] - 1) if adjust else (shape["x"], shape["y"])
+            )
+            if not res:
+                assert ds.transform is not None
+                transform = ds.transform * Affine.scale(1 / shape_x, 1 / shape_y)
+        else:
+            return None, None
+        return (int(shape_y), int(shape_x)), transform
+
+    # We handle open_datafiles first since it should only be set when it's absolutely necessary
+    if open_datafiles:
+        bands = {n: BandInfo(ds, n) for n in list(ds.product.measurements)}
```

Comment:
No need for this code branch.  Higher level code should getting transform/shape, crs, and extent directly from the data.